src/dasi/shelf/ShelfCatalogue.cc

```diff
@@ -0,0 +1,74 @@
+
+#include "dasi/shelf/ShelfCatalogue.h"
+
+#include "dasi/api/Config.h"
+
+#include "dasi/core/FileReadHandle.h"
+#include "dasi/core/SplitReferenceKey.h"
+
+#include <cctype>
+#include <fstream>
+#include <iomanip>
+#include <sstream>
+
+namespace dasi::shelf {
+
+static std::string encode(const std::string_view& v) {
+    std::ostringstream r;
+    r.fill('0');
+    r << std::hex;
+
+    for (const auto c : v) {
+        auto cu = static_cast<unsigned char>(c);
```

Comment:
why are we doing this?

Reply:
The behaviour of isalnum() (and other similar is* functions) is undefined if the input (type int) is not in the range 0-255
https://en.cppreference.com/w/cpp/string/byte/isalnum